### ISSUE TITLE
[0.4.5] react/prop-types ignores the Props type argument of React.forwardRef

### ISSUE THREAD

**Issue by fireairforce** [utooland/utoo-lint#1276]
## Summary

A component with React.forwardRef<HTMLDivElement, Props> is reported as missing props validation despite the generic Props argument. Both namespace and named imports reproduce. An explicit callback parameter annotation is recognized, while the untyped control correctly reports.

Verified with @utoo/lint 0.4.5, ESLint 8.57.1, @typescript-eslint/parser and eslint-plugin 6.21.0, eslint-plugin-react 7.37.5, Node.js 22.23.1 on macOS arm64. Each example enables only the indicated rule, with no parsing failures or suppression comments.

## Setup

```sh
npm init -y
npm install --save-dev @utoo/lint@0.4.5 eslint@8.57.1 @typescript-eslint/parser@6.21.0 @typescript-eslint/eslint-plugin@6.21.0 eslint-plugin-react@7.37.5 typescript@5.3.3
```

`tsconfig.json`:

```json
{"compilerOptions":{"strict":true,"target":"ES2022","jsx":"react"},"include":["example.tsx"]}
```

Run each case separately, replacing example.tsx and both config files with the corresponding contents below:

```sh
./node_modules/.bin/eslint --no-eslintrc --config .eslintrc.cjs --format=json example.tsx
./node_modules/.bin/utlint --config utlint.config.json --format=json example.tsx
```

## ref-generic

`example.tsx`:

```tsx
import React from "react";type Props={name:string};export const Example=React.forwardRef<HTMLDivElement,Props>((props,ref)=><div ref={ref}>{props.name}</div>);
```

`.eslintrc.cjs`:

```js
module.exports = {
  "root": true,
  "parser": "@typescript-eslint/parser",
  "parserOptions": {
    "ecmaVersion": 2022,
    "sourceType": "module",
    "ecmaFeatures": {
      "jsx": true
    }
  },
  "plugins": [
    "react"
  ],
  "rules": {
    "react/prop-types": "error"
  },
  "settings": {
    "react": {
      "version": "18.2"
    }
  }
};
```

`utlint.config.json`:

```json
{
  "rules": {
    "react/prop-types": "error"
  },
  "settings": {
    "react": {
      "version": "18.2"
    }
  }
}
```

ESLint: 0 diagnostic(s). utlint: 1 diagnostic(s).

eslint messages:

```text
(none)
```

utlint messages:

```text
'name' is missing in props validation
```

## ref-named-import

`example.tsx`:

```tsx
import {forwardRef} from "react";type Props={name:string};export const Example=forwardRef<HTMLDivElement,Props>((props,ref)=><div ref={ref}>{props.name}</div>);
```

`.eslintrc.cjs`:

```js
module.exports = {
  "root": true,
  "parser": "@typescript-eslint/parser",
  "parserOptions": {
    "ecmaVersion": 2022,
    "sourceType": "module",
    "ecmaFeatures": {
      "jsx": true
    }
  },
  "plugins": [
    "react"
  ],
  "rules": {
    "react/prop-types": "error"
  },
  "settings": {
    "react": {
      "version": "18.2"
    }
  }
};
```

`utlint.config.json`:

```json
{
  "rules": {
    "react/prop-types": "error"
  },
  "settings": {
    "react": {
      "version": "18.2"
    }
  }
}
```

ESLint: 0 diagnostic(s). utlint: 1 diagnostic(s).

eslint messages:

```text
(none)
```

utlint messages:

```text
'name' is missing in props validation
```

## ref-annotated

`example.tsx`:

```tsx
import React from "react";type Props={name:string};export const Example=React.forwardRef((props:Props,ref)=><div ref={ref}>{props.name}</div>);
```

`.eslintrc.cjs`:

```js
module.exports = {
  "root": true,
  "parser": "@typescript-eslint/parser",
  "parserOptions": {
    "ecmaVersion": 2022,
    "sourceType": "module",
    "ecmaFeatures": {
      "jsx": true
    }
  },
  "plugins": [
    "react"
  ],
  "rules": {
    "react/prop-types": "error"
  },
  "settings": {
    "react": {
      "version": "18.2"
    }
  }
};
```

`utlint.config.json`:

```json
{
  "rules": {
    "react/prop-types": "error"
  },
  "settings": {
    "react": {
      "version": "18.2"
    }
  }
}
```

ESLint: 0 diagnostic(s). utlint: 0 diagnostic(s).

eslint messages:

```text
(none)
```

utlint messages:

```text
(none)
```

## ref-untyped

`example.tsx`:

```tsx
import React from "react";export const Example=React.forwardRef((props,ref)=><div ref={ref}>{props.name}</div>);
```

`.eslintrc.cjs`:

```js
module.exports = {
  "root": true,
  "parser": "@typescript-eslint/parser",
  "parserOptions": {
    "ecmaVersion": 2022,
    "sourceType": "module",
    "ecmaFeatures": {
      "jsx": true
    }
  },
  "plugins": [
    "react"
  ],
  "rules": {
    "react/prop-types": "error"
  },
  "settings": {
    "react": {
      "version": "18.2"
    }
  }
};
```

`utlint.config.json`:

```json
{
  "rules": {
    "react/prop-types": "error"
  },
  "settings": {
    "react": {
      "version": "18.2"
    }
  }
}
```

ESLint: 1 diagnostic(s). utlint: 1 diagnostic(s).

eslint messages:

```text
'name' is missing in props validation
```

utlint messages:

```text
'name' is missing in props validation
```
